From 7dfca8531d76132c9e92563837b93e84f020fc77 Mon Sep 17 00:00:00 2001
From: Gabe Black <gabe.black@gmail.com>
Date: Fri, 25 Feb 2022 23:11:18 -0800
Subject: [PATCH 298/757] cpu: Remove an unused variable from one of the branch
 predictors.

The variable upsets clang and breaks the build.

Change-Id: Ia2315e1753b6e9e701bf18eadf8cf448643577b4
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/57172
Maintainer: Gabe Black <gabe.black@gmail.com>
Reviewed-by: Daniel Carvalho <odanrc@yahoo.com.br>
Tested-by: kokoro <noreply+kokoro@google.com>
---
 src/cpu/pred/multiperspective_perceptron_tage.cc | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/cpu/pred/multiperspective_perceptron_tage.cc b/src/cpu/pred/multiperspective_perceptron_tage.cc
index 67c470f..6176d9c 100644
--- a/src/cpu/pred/multiperspective_perceptron_tage.cc
+++ b/src/cpu/pred/multiperspective_perceptron_tage.cc
@@ -328,11 +328,9 @@ MPP_StatisticalCorrector::gUpdate(Addr branch_pc, bool taken, int64_t hist,
                    int nbr, int logs, std::vector<int8_t> & w,
                    StatisticalCorrector::BranchInfo* bi)
 {
-    int percsum = 0;
     for (int i = 0; i < nbr; i++) {
         int64_t bhist = hist & ((int64_t) ((1 << length[i]) - 1));
         int64_t index = gIndex(branch_pc, bhist, logs, nbr, i);
-        percsum += (2 * tab[i][index] + 1);
         ctrUpdate(tab[i][index], taken, scCountersWidth - (i < (nbr - 1)));
     }
 }
-- 
1.8.3.1

